src/Catalyst.Core.Lib/IO/Events/TransactionReceivedEvent.cs

```diff
@@ -59,10 +64,23 @@ public class TransactionReceivedEvent : ITransactionReceivedEvent
         public ResponseCode OnTransactionReceived(ProtocolMessage protocolMessage)
         {
             var transactionBroadcast = protocolMessage.FromProtocolMessage<TransactionBroadcast>();
-            var transactionValid = _validator.ValidateTransaction(transactionBroadcast.PublicEntry);
-            if (!transactionValid)
+            PublicEntry publicEntry = transactionBroadcast.PublicEntry;
+            if (publicEntry.SenderAddress.Length == 32)
             {
-                return ResponseCode.Error;
+                var transactionValid = _validator.ValidateTransaction(publicEntry);
+                if (!transactionValid)
+                {
+                    return ResponseCode.Error;
+                }
+                
+                byte[] kvmAddressBytes = Keccak.Compute(publicEntry.SenderAddress.ToByteArray()).Bytes.AsSpan(12).ToArray();
+                string hex = kvmAddressBytes.ToHexString() ?? throw new ArgumentNullException("kvmAddressBytes.ToHexString()");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name 'kvmAddressBytes.ToHexString()' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=5051857)